Fixup CioEncoder to gracefully handle unpickling in situation where t…

oxtopus opened the pull request:
…he cacheDir differs between original location and new

BoltzmannBrain commented:
@oxtopus running imbu.py fails (from projects/nlp/):
```
$ python ../../htmresearch/frameworks/nlp/imbu.py --dataPath data/sample_reviews/sample_reviews.csv -m CioDocumentFingerprint
Traceback (most recent call last):
  File "../../htmresearch/frameworks/nlp/imbu.py", line 434, in <module>
    main()
  File "../../htmresearch/frameworks/nlp/imbu.py", line 407, in main
    networkConfigName=args.networkConfigName)
  File "../../htmresearch/frameworks/nlp/imbu.py", line 256, in createModel
    **modelFactoryKwargs)
  File "../../htmresearch/frameworks/nlp/imbu.py", line 202, in _modelFactory
    model = createModel(modelName, **kwargs)
  File "/Users/alavin/nta/nupic.research/htmresearch/frameworks/nlp/model_factory.py", line 119, in createModel
    return getattr(ClassificationModelTypes, modelName)(**kwargs)
  File "/Users/alavin/nta/nupic.research/htmresearch/frameworks/nlp/classify_fingerprint.py", line 67, in __init__
    apiKey=apiKey)
  File "/Users/alavin/nta/nupic.research/htmresearch/encoders/cio_encoder.py", line 84, in __init__
    self.cacheDir = cacheDir
AttributeError: can't set attribute
```

oxtopus commented:
@BoltzmannBrain Thanks.  Fixed, but you'll need to re-pull to test since I had to unmerge https://github.com/numenta/nupic.research/pull/459 i.e. I rewrote history.

BoltzmannBrain commented:
:+1:
